### PR TITLE
Integrate Apollo configuration

### PR DESCRIPTION
**Pull request by sieniven**
## Summary

- Add Apollo client into Erigon node to support apollo configuration management
- Add support for jsonrpc, sequencer and l2gaspricer prefix namespace pipelines for config management of these services
- Add callbacks to load of node and eth configs from apollo config changes
- Add support for dynamic config changes with apollo config singleton instances on ethconfig and nodeconfig
- Depends on refactors from PR #53

### CODE REVIEW

File: eth/ethconfig/apollo_xlayer.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+type ApolloConfig struct {
+	EnableApollo bool
+	Conf         Config
+	sync.RWMutex
+}
+
+var apolloConfig = &ApolloConfig{}
+
+// GetApolloConfig returns a copy of the singleton instance apollo config
+func GetApolloConfig() (Config, error) {
```
Comment (giskook) [inline eth/ethconfig/apollo_xlayer.go:18]: It seems `GetApolloConfig` is not used? We just update the `apolloConfig`?
Some we need dynmic adjust praramters are list below:
* seq: https://github.com/okx/xlayer-node/blob/dev/sequencer/apollo_xlayer.go#L11
* jsonrpc: https://github.com/okx/xlayer-node/blob/dev/jsonrpc/apollo_xlayer.go#L10
* l2gaspricer: -
Some features are not migrated, but I think we can add jsonrpc to enable batch request and dynmic change batch request's limit.

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -621,6 +621,23 @@ var (
 		Name:  "debug.step-after",
 		Usage: "Start incrementing by debug.step after this block",
 	}
+	// XLayer apollo
+	ApolloEnableFlag = cli.BoolFlag{
+		Name:  "zkevm.apollo-enable",
```
[the remark this answers is not in the document]
Comment (sieniven) [reply cmd/utils/flags.go]: Resolved in d4f2493

File: eth/ethconfig/config_xlayer.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package ethconfig
+
+type XLayerConfig struct {
```
[the remark this answers is not in the document]
Comment (sieniven) [reply eth/ethconfig/config_xlayer.go:4]: Resolved in d4f2493

File: eth/ethconfig/apollo_xlayer.go (hunk 1 above)
Comment (sieniven) [reply eth/ethconfig/apollo_xlayer.go:18]: Yup it is not used for now. But in the future in pipelines where apollo configs are implemented, we need to ensure memory safety and not use the singleton instance directly. Thus, this safe function is defined here that encapsulates handling the memory safety, and returns a copy of the apollo config.

File: zk/apollo/apollo.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	nsMap := make(map[string]string)
+	namespaces := strings.Split(ethCfg.Zk.XLayer.Apollo.NamespaceName, ",")
+	for _, namespace := range namespaces {
+		prefix, err := getNamespacePrefix(namespace)
+		if err != nil {
+			utils.Fatalf("failed init apollo: %v", err)
+		}
+		nsMap[prefix] = namespace
```
Comment (giskook) [inline zk/apollo/apollo.go:57]: It seems some namespace will be overwritted? eg.  if both `jsonrpc-ro` and `jsonrpc-exp` exists, I think one of them will be overwritted.

Comment (sieniven) [reply zk/apollo/apollo.go]: Resolved in 08a36786163728426612d9cc9cb2654da9a9520d

**Review by giskook — APPROVED** [review]

File: zk/apollo/apollo.go (hunk 4 above)
Comment (sieniven) [reply zk/apollo/apollo.go:62]: 08a36786163728426612d9cc9cb2654da9a9520d should contain the correct fixes. We should not accept duplicate prefix namespaces in the node configuration, and will throw an error on node initialization if the configs were set incorrectly.

**Pull request merged**